src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -452,15 +452,18 @@ class Direct$Type$Buffer$RW$$BO$
         checkSession();
 
         int length = end - start;
+        Objects.checkFromIndexSize(start, length,
+                                   csq == null ? 0 : csq.length());
```

Comment:
```suggestion
        Objects.checkFromIndexSize(start, length, csq.length());
```
Aren't the calls to `appendChars` already guarded by `instanceof` checks, which produces `false` for `null`?

Reply:
Oh, I think you are correct. Removed in 52c9458.